### DISCUSSION TITLE
Convert python@3.9 from source to bottle?

### DISCUSSION THREAD

**Discussion by jimtut** [Homebrew/discussions#453] [Everyday usage] [answered]
I have python@3.9 installed on my M1 Mac, but I did it early and had to build from source.  Every time I do "brew upgrade" and something that depends on python gets upgraded, my python gets upgraded too, building from source again.  (Maybe the Python source really is changing that fast, not sure, but it happens a LOT.)

How can I switch my install from source to bottle, so it will upgrade faster than building from source?  I'm afraid to uninstall it (not sure it would, given that other SW depends on it) as I don't want to lose the libraries I've installed in /opt/homebrew/lib/python3.9/site-packages.

**Comment by gromgit** [comment, marked as answer]
Try `brew reinstall --force-bottle python@3.9`.

**Reply by jimtut** [reply]
Ah, `reinstall`, I missed that, thanks!

**Comment by carlocab** [comment]
Do you have the CLT installed? The Python formula won't pour a bottle (unless forced) if you don't have the CLT installed. `brew config` will tell you if you have it installed; just look for the `CLT:` line. (The CLT that comes with the Xcode app doesn't count.)

I ask this because I've never needed to switch formulae that I've built from source to bottles; `brew upgrade` just does that on its own.

**Reply by jimtut** [reply]
Weird, I did NOT have CLT, although I was able to pour bottles.  I thought I re-installed CLT after 11.1 upgrade, but maybe there are 2 versions, and I did the Xcode version before?  I just did the "sudo xcode-select --install" command, and that gets "brew config" to recognize the CLT.

**Reply by carlocab** [reply]
Yea, upgrading to 11.1 wipes out the CLT for some reason. The latest version of `brew` should now have slightly more informative error messages when it doesn't want to pour a bottle.
